[BACKLOG-37004] Page Zoom (AA) - File Open Dialog and Base GWT and XUL/GWT Dialog Classes

dcleao opened the pull request:
Merge with:
- XUL
- Platform

buildguy commented:
### :white_check_mark: Build finished in 6m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **359**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-839/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

dcleao commented:
Rebased on `wcag` to resolve recent conflicts.

buildguy commented:
### :white_check_mark: Build finished in 5m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **359**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-839/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

buildguy commented:
### :white_check_mark: Build finished in 5m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **359**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-839/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

soagarwal1 commented:
Except for a few Sonar code smells, commit looks good to be merged.
As discussed, putting the responsive field is being conservative for now and may/may not be required later on once we tackle different dialogs individually. Currently, we need to set the values for individual Dialogs

soagarwal1 reviewed (changes requested):
Except for a few Sonar code smells, commit looks good to be merged.
As discussed, putting the responsive field is being conservative for now and may/may not be required later on once we tackle different dialogs individually. Currently, we need to set the values for individual Dialogs.

buildguy commented:
### :warning: Build finished in 5m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.gwt.widgets.client.dialogs.DialogBoxTest.testSetFocusWidget</b> (click to expand)</summary><p>

```

Wanted but not invoked:
focusWidget.setFocus(true);
-> at org.pentaho.gwt.widgets.client.dialogs.DialogBoxTest.testSetFocusWidget(DialogBoxTest.java:57)
Actually, there were zero interactions with this mock.

```

</p></details>

  
Tests run: **359**, Failures: **1**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-839/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

buildguy commented:
### :white_check_mark: Build finished in 5m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **359**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-839/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

dcleao commented:
@pentaho/wcag, please re-review the **3 new commits**. The changes are:
1. Rebased on latest of `wcag`
2. Addressed some of the Sonar issues
3. Retreated on _some_ of the glass pane changes that I started second-guessing were actual improvements — I may later attempt improving this, again, but did not want to compromise merging further.
4. Fixed an issue I realized I would be introducing due affecting `doAutoFocus()` (initial focus). Must be called explicitly on `center()`, also.
5. Fixed issues with the proposed responsive dialogs CSS which were already causing problems, when trying to use them for the BootstrapJS dialogs. Button section now wraps instead of scrolling horizontally. This approach seems more robust in face of small dialog widths, many buttons and possibly long button text.

Before actually merging, I will re-write history to get rid of the new 3 commits and still preserve the original commit structure.

dcleao commented:
Rewrote history to the minimum number of commits and updated the commits text where appropriate. No changes in code.

buildguy commented:
### :white_check_mark: Build finished in 5m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **359**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-839/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_
